File: 12/12.php
<?php

function divisors($x) {
  $divs = 0;
  $y = sqrt($x);
  for ($i = 1; $i <= $y; $i++) {
    if (!($x % $i)) $divs += 2;
  }
  return $divs + 1;
}

$tri = 0;
for ($i = 1; divisors($tri) <= 500; $i++) {
  $tri += $i;
}
print $tri."\n";

?>

File: 12/12.nim
import math

proc divisors(x: int): int =
  var divs = 0
  let y = sqrt(float(x))
  for i in 1..int(y):
    if x mod i == 0:
      divs += 2
  divs + 1

var
  tri = 0
  i = 1
while divisors(tri) <= 500:
  tri += i
  i += 1
echo tri
